gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixJob.java

```diff
@@ -65,28 +70,32 @@ public void executeImpl(JobExecutionContext context) throws JobExecutionExceptio
 
     try {
       final JobLauncher jobLauncher = new GobblinHelixJobLauncher(jobProps, helixManager, appWorkDir, eventMetadata);
-
       if (Boolean.valueOf(jobProps.getProperty(GobblinClusterConfigurationKeys.JOB_EXECUTE_IN_SCHEDULING_THREAD,
-          Boolean.toString(GobblinClusterConfigurationKeys.JOB_EXECUTE_IN_SCHEDULING_THREAD_DEFAULT)))) {
+              Boolean.toString(GobblinClusterConfigurationKeys.JOB_EXECUTE_IN_SCHEDULING_THREAD_DEFAULT)))) {
         jobScheduler.runJob(jobProps, jobListener, jobLauncher);
       } else {
-        // if not executing in the scheduling thread then submit a runnable to the job scheduler's ExecutorService
-        // for asynchronous execution.
-        Runnable runnable = new Runnable() {
-          @Override
-          public void run() {
-            try {
-              jobScheduler.runJob(jobProps, jobListener, jobLauncher);
-            } catch (JobException je) {
-              log.error("Failed to run job " + jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY), je);
-            }
-          }
-        };
-
-        jobScheduler.submitRunnableToExecutor(runnable);
+        cancellable = jobScheduler.submitJob(jobProps, jobListener, jobLauncher);
       }
     } catch (Throwable t) {
       throw new JobExecutionException(t);
     }
   }
+
+  @Override
+  public void interrupt() throws UnableToInterruptJobException {
```

Comment:
That isn't really my take of how Quartz interrupt is designed.  They specifically call out on the FAQ: 
> ## How do I stop a Job that is currently executing?
> See the org.quartz.InterruptableJob interface, and the Scheduler.interrupt(String, String) method.

Additionally, in the current Gobblin implementation, `interrupt` is called when shutting down `JobScheduler`.  The issue is that the jobs don't implement `interrupt` so nothing happens.  

I can add a boolean that gets set when the JobScheduler starts shutting down.  The Jobs can check for that bool when `interrupt` is called and cancel only if that is set.